cognite/client/_api/files.py

```diff
@@ -767,12 +769,30 @@ def download(
                 >>> c = CogniteClient()
                 >>> c.files.download(directory="my_directory", id=[1,2,3], external_id=["abc", "def"])
         """
+
         if isinstance(directory, str):
             directory = Path(directory)
+        assert directory.is_dir(), f"{directory} is not a directory"
+
         all_ids = IdentifierSequence.load(id, external_id).as_dicts()
         id_to_metadata = self._get_id_to_metadata_map(all_ids)
-        assert directory.is_dir(), f"{directory} is not a directory"
-        self._download_files_to_directory(directory, all_ids, id_to_metadata)
+
+        if keep_folder_structure:
+            for metadata in id_to_metadata.values():
```

Comment:
Doing these one by one is going to be significantly slower than the other implementation. Let's download them in parallel, and then create the necessary subdural, ane then move the files to the appropriate path instead.